interface/patient_file/report/custom_report.php

```diff
@@ -624,27 +666,43 @@ function postToGet($arin)
                         echo "</div></div>\n"; // HTML to PDF conversion will fail if there are open tags.
                         $content = getContent();
                         $pdf->writeHTML($content); // catch up with buffer.
-                        $pg_header = "<span>" . xlt('Document') . " " . text($fname) ."</span>";
-                        $tempDocC = new C_Document;
-                        $pdfTemp = $tempDocC->retrieve_action($d->get_foreign_id(), $document_id, false, true, true, true);
-                        // tmp file in temporary_files_dir
-                        $from_file_tmp_name = tempnam($GLOBALS['temporary_files_dir'], "oer");
-                        file_put_contents($from_file_tmp_name, $pdfTemp);
-                        $pagecount = $pdf->setSourceFile($from_file_tmp_name);
-                        for ($i = 0; $i < $pagecount; ++$i) {
-                            $pdf->AddPage();
-                            $itpl = $pdf->importPage($i+1);
-                            $pdf->useTemplate($itpl);
+                        $err = '';
+                        try {
+                            // below header isn't being used. missed maybe!
+                            $pg_header = "<span>" . xlt('Document') . " " . text($fname) ."</span>";
+                            $tempDocC = new C_Document;
+                            $pdfTemp = $tempDocC->retrieve_action($d->get_foreign_id(), $document_id, false, true, true, true);
+                            // tmp file in temporary_files_dir
+                            $from_file_tmp_name = tempnam($GLOBALS['temporary_files_dir'], "oer");
+                            file_put_contents($from_file_tmp_name, $pdfTemp);
+
+                            $pagecount = $pdf->setSourceFile($from_file_tmp_name);
```

Comment:
@sjpadgett ,
Is this the line where the exception is thrown here or is it further above?

Reply:
@bradymiller $pdf->setSourceFile will throw if setasign FPDI free parser can't parse. Could get around if we wanted to require GhostScript but issue there is similar to wkhtmltopdf exe. I could do this but GhostScript is AGNU  licence(is that a problem?).

I'm still not sure this is best approach from the clinical workflow POV but plan to circle back to this so I can fax all PDFs from zip.

Reply:
Check out my comment on the patch PR. Best to avoid AGPL licensed code. Real bummer that php can't support more recent versions. Hopefully this gets addressed at some point.